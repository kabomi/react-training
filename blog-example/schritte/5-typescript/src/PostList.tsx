import React from "react";
import moment from "moment";
import { BlogPost } from "./types";

function formattedDate(date: string) {
  return moment(date).format("DD.MM.YYYY");
}

type BlogListProps = {
  posts: BlogPost[];
  onAddPost(): void;
};

export default function BlogList(props: BlogListProps) {
  const posts = props.posts;

  return (
    <>
      <button onClick={props.onAddPost}>Add Post</button>
      {posts.map(p => (
        <article key={p.id} className="Container">
          <p className="Date">{formattedDate(p.date)}</p>
          <h1>{p.title}</h1>
          <p>{p.body}</p>
        </article>
      ))}
    </>
  );
}
