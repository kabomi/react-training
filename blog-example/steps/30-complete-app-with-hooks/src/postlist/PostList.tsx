import React from "react";
import { Link } from "react-router-dom";
import { linkToPost, formattedDate } from "../utils";

type PostListProps = {
  posts: Array<{
    id: string;
    date: string;
    title: string;
    teaser: string;
  }>;
};

export default function PostList(props: PostListProps) {
  const posts = props.posts;

  return (
    <>
      <Link className="Button" to="/add">
        Add Post
      </Link>
      {posts.map(p => (
        <Link key={p.id} to={linkToPost(p)}>
          <article className="Container">
            <p className="Date">{formattedDate(p.date)}</p>
            <h1>{p.title}</h1>
            {p.teaser} <span>Read more</span>
          </article>
        </Link>
      ))}
    </>
  );
}
