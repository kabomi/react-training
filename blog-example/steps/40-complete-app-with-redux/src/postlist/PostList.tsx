import React from "react";
import { Link } from "react-router-dom";
import { linkToPost, formattedDate } from "../utils";
import { useAuth } from "auth/AuthContext";

type PostListProps = {
  posts: Array<{
    id: string;
    date: string;
    title: string;
    teaser: string;
    userId: string;
  }>;
};

function UserBadge() {
  return <div className="UserBadge">Your Post!</div>;
}

export default function PostList({ posts }: PostListProps) {
  const { authState } = useAuth();
  const currentUserId = authState?.userId;

  return (
    <>
      <Link className="Button" to="/add">
        Add Post
      </Link>
      {posts.map(p => (
        <Link key={p.id} to={linkToPost(p)}>
          <article className="Container">
            <p className="Date">{formattedDate(p.date)}</p>
            <header>
              <h1>{p.title}</h1>
              {currentUserId === p.userId && <UserBadge />}
            </header>
            {p.teaser} <span>Read more</span>
          </article>
        </Link>
      ))}
    </>
  );
}
