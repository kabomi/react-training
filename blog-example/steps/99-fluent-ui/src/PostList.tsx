import { CommandBar, FontSizes, Separator, Stack, Text } from "@fluentui/react";
import { Depths, NeutralColors } from "@uifabric/fluent-theme";
import moment from "moment";
import React from "react";
import { BlogPost } from "./types";

function formattedDate(date: string) {
  return moment(date).format("DD.MM.YYYY");
}

type PostListProps = {
  posts: BlogPost[];
  onAddPost(): void;
};

export default function PostList(props: PostListProps) {
  const posts = props.posts;

  return (
    <Stack
      tokens={{
        childrenGap: "30"
      }}
      style={{
        background: NeutralColors.gray20
      }}
    >
      <CommandBar
        items={[
          {
            key: "addPost",
            text: "Add Post",
            iconProps: { iconName: "Add" },
            onClick: props.onAddPost
          }
        ]}
      />
      {posts.map(post => (
        <Stack key={post.id}>
          <Stack
            style={{
              boxShadow: Depths.depth8,
              background: NeutralColors.white,
              padding: FontSizes.xLarge
            }}
          >
            <Text>{formattedDate(post.date)}</Text>
            <Separator />
            <Text variant="xxLarge">{post.title}</Text>
            {post.body.split("\\n").map((p, ix) => (
              <Text key={ix} variant="large">
                {p}
              </Text>
            ))}
          </Stack>
        </Stack>
      ))}
    </Stack>
  );
}
