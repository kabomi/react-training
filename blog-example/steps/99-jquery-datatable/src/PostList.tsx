import React from "react";
import moment from "moment";
import { BlogPost } from "./types";
function formattedDate(date: string) {
  return moment(date).format("DD.MM.YYYY");
}

type PostListProps = {
  posts: BlogPost[];
  onAddPost(): void;
};

export default function PostList(props: PostListProps) {
  const posts = props.posts;
  const tableRef = React.useRef<HTMLTableElement>(null);

  React.useEffect(() => {
    if (props.posts.length === 0) {
      return;
    }
    let cleanup: any;
    if (tableRef.current) {
      const table = $(tableRef.current).DataTable({ colReorder: true });
      cleanup = function() {
        table.destroy();
      };
    }
    return cleanup;
  }, [props.posts]);

  return (
    <>
      <button onClick={props.onAddPost}>Add Post</button>
      <table ref={tableRef}>
        <thead>
          <tr>
            <th>Date</th>
            <th>Title</th>
          </tr>
        </thead>
        <tbody>
          {posts.map(p => (
            <tr key={p.id}>
              <td>{formattedDate(p.date)}</td>
              <td>{p.title}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
}
