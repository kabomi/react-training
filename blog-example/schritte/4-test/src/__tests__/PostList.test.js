import React from "react";
import renderer from "react-test-renderer";

import { render, fireEvent } from "@testing-library/react";

import PostList from "../PostList";

const mockPosts = [
  { id: "1", title: "One", body: "Lorem ipsum" },
  { id: "2", title: "Second Post", body: "Some more content" }
];

it("renders correctly", () => {
  const tree = renderer.create(<PostList posts={mockPosts} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("invokes callback on button click", () => {
  const onAddPostFn = jest.fn();

  // Render
  const { getByText } = render(<PostList onAddPost={onAddPostFn} posts={mockPosts} />);

  // search the button
  const buttonElement = getByText(/Add Post/i);

  // click the button
  fireEvent.click(buttonElement);

  // make sure it has been invoked
  expect(onAddPostFn).toHaveBeenCalled();
});
