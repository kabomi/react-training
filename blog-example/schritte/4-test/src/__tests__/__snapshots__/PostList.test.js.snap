// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <button>
    Add Post
  </button>,
  <article
    className="Container"
  >
    <p
      className="Date"
    >
      05.01.2020
    </p>
    <h1>
      One
    </h1>
    <p>
      Lorem ipsum
    </p>
  </article>,
  <article
    className="Container"
  >
    <p
      className="Date"
    >
      05.01.2020
    </p>
    <h1>
      Second Post
    </h1>
    <p>
      Some more content
    </p>
  </article>,
]
`;
